<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BaseMainWindow</class>
 <widget class="QMainWindow" name="BaseMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>g2o Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QFrame" name="frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QCheckBox" name="cbDrawAxis">
         <property name="text">
          <string>Draw Axis</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string># Iterations</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spIterations">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbRobustKernel">
         <property name="text">
          <string>Robust Kernel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbOnlyLoop">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Only apply the robust kernel for loop closures</string>
         </property>
         <property name="text">
          <string>Non Sequential</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="coRobustKernel">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Kernel Width</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="leKernelWidth">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>1.0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Optimizer</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="coOptimizer"/>
       </item>
       <item>
        <widget class="QPushButton" name="btnOptimizerParamaters">
         <property name="toolTip">
          <string>Adjust the parameters of the optimizer</string>
         </property>
         <property name="text">
          <string>Parameters</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>15</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QComboBox" name="cbxIniitialGuessMethod">
         <property name="toolTip">
          <string>Which method is applied to compute the initial guess</string>
         </property>
         <item>
          <property name="text">
           <string>Spanning Tree</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Odometry</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnInitialGuess">
         <property name="text">
          <string>Initial Guess</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnSetZero">
         <property name="text">
          <string>SetZero</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnOptimize">
         <property name="text">
          <string>Optimize</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnForceStop">
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnQuit">
         <property name="text">
          <string>Quit</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>364</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="g2o::G2oQGLViewer" name="viewer" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>100</verstretch>
        </sizepolicy>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="plainTextEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionSave_Screenshot"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_Viewer_State"/>
    <addaction name="actionSave_Viewer_State"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionWhite_Background"/>
    <addaction name="actionDefault_Background"/>
    <addaction name="separator"/>
    <addaction name="actionDump_Images"/>
    <addaction name="separator"/>
    <addaction name="actionProperties"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionWhite_Background">
   <property name="text">
    <string>White Background</string>
   </property>
  </action>
  <action name="actionDefault_Background">
   <property name="text">
    <string>Default Background</string>
   </property>
  </action>
  <action name="actionDump_Images">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Dump Images</string>
   </property>
  </action>
  <action name="actionProperties">
   <property name="text">
    <string>Draw Options</string>
   </property>
  </action>
  <action name="actionSave_Screenshot">
   <property name="text">
    <string>Save Screenshot</string>
   </property>
  </action>
  <action name="actionSave_Viewer_State">
   <property name="text">
    <string>Save Viewer State</string>
   </property>
  </action>
  <action name="actionLoad_Viewer_State">
   <property name="text">
    <string>Load Viewer State</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>g2o::G2oQGLViewer</class>
   <extends>QWidget</extends>
   <header>g2o_qglviewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>cbRobustKernel</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_4</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbRobustKernel</sender>
   <signal>toggled(bool)</signal>
   <receiver>leKernelWidth</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>147</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnQuit</sender>
   <signal>clicked()</signal>
   <receiver>BaseMainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>541</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>554</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbRobustKernel</sender>
   <signal>toggled(bool)</signal>
   <receiver>coRobustKernel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>56</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbRobustKernel</sender>
   <signal>toggled(bool)</signal>
   <receiver>cbOnlyLoop</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>160</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
